### DISCUSSION TITLE
Populating payloads of enum instances?

### DISCUSSION THREAD

**Discussion by NSExceptional** [Azoy/Echo#39] [Q&A] [answered]
`EnumDescriptor` doesn't appear to tell me anything about how payloads are stored. I assume it works similar to any, where values are only inline up to a certain point. Is this part of the ABI not stable or something?

Edit: looking at [TypeLayout.rst](https://github.com/apple/swift/blob/main/docs/ABI/TypeLayout.rst) tells me more about how enums are laid out by the compiler, but not how to fetch that layout for any given enum at runtime, such as for enums with multiple values in a single payload, or enums with more than one payload case.

Edit 2: [Runtime.md](https://github.com/apple/swift/blob/main/docs/Runtime.md) also looks promising

**Comment by NSExceptional** [comment, marked as answer]
Answer here: https://forums.swift.org/t/populating-accessing-the-payload-of-an-enum-case-at-runtime/47921/2

**Reply by NSExceptional** [reply]
@Azoy Are enums with raw values treated as payload cases where the payload is exposed via `.rawValue`?

**Reply by Azoy** [reply]
No, enums with raw values are treated as empty cases.
```swift
enum Color: Int {
  case blue = 123
}

// is really

enum Color: RawRepresentable {
  typealias RawValue = Int

  case blue

  var rawValue: Int {
    switch self {
    case .blue:
      return 123
    }
  }
}
```
